### DISCUSSION TITLE
Middleware is reloaded on every request in v13 but it wasn't in v12.2

### DISCUSSION THREAD

**Discussion by NOtherDev** [vercel/next.js#44053] [Help]
We're experiencing a behavior change for middlewares between v12.2 and v13.

In development, in v12.2 middlewares were initialized once and the same function instance handled all the subsequent requests (at least for some time).
After upgrading to v13 we can see that each request is being handled by the new instance of the middleware.
It breaks our case in which we're keeping some short-lived caches in-memory of the middleware module. Right now it's ineffective as the module is never reused.

Pseudocode:
```
console.log('initializing middleware')

export async function middleware(request: NextRequest): Promise<NextResponse> {
  console.log('in middleware', request.nextUrl.pathname)
}
```

Log excerpt from v12.2
```
initializing middleware
in middleware /page1
in middleware /page2
in middleware /page3
in middleware /page4
```

Log excerpt from v13
```
initializing middleware
in middleware /page1
initializing middleware
in middleware /page2
initializing middleware
in middleware /page3
initializing middleware
in middleware /page4
```

Was this change expected? 
Is there a way to restore the previous behavior?

**Comment by jvmusin** [comment]
I have the same problem and I can't find any information about that at all.

**Comment by mgreenw** [comment]
+1 with the same issue. Expected behavior is that middleware should only be initialized once globally (until modified in dev), but it seems to be initializing once per request.

Issue: https://github.com/vercel/next.js/issues/43999
